Content.Client/_White/VisionLimit/Systems/VisionLimitSystem.cs

```diff
@@ -0,0 +1,126 @@
+﻿using Content.Client._White.VisionLimit.Overlays;
+using Content.Shared._White.VisionLimit.Components;
+using Content.Shared.Clothing;
+using Content.Shared.Inventory.Events;
+using Robust.Client.Graphics;
+using Robust.Client.Player;
+using Robust.Shared.Player;
+
+namespace Content.Client._White.VisionLimit.Systems;
+
+public sealed class VisionLimitSystem : EntitySystem
+{
+    [Dependency] private readonly IOverlayManager _overlayMan = default!;
+    [Dependency] private readonly IPlayerManager _player = default!;
+
+    private VisionLimitOverlay _overlay = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<VisionLimitComponent, ComponentInit>(OnCompInit);
+        SubscribeLocalEvent<VisionLimitComponent, ComponentShutdown>(OnCompShutdown);
+
+        SubscribeLocalEvent<VisionLimitComponent, LocalPlayerAttachedEvent>(OnPlayerAttached);
+        SubscribeLocalEvent<VisionLimitComponent, LocalPlayerDetachedEvent>(OnPlayerDetached);
+
+        SubscribeLocalEvent<ClothingLimitVisionComponent, GotEquippedEvent>(OnGotEquipped);
+        SubscribeLocalEvent<ClothingLimitVisionComponent, GotUnequippedEvent>(OnGotUnequipped);
+        SubscribeLocalEvent<ClothingLimitVisionComponent, ItemMaskToggledEvent>(OnMaskToggled);
+
+        _overlay = new();
+        _overlay.ZIndex = -2; // Draw under damage overlay etc.
+    }
+
+    // Player entity interactions
+
+    private void OnPlayerAttached(EntityUid uid, VisionLimitComponent component, LocalPlayerAttachedEvent args)
+    {
+        UpdateOverlay(component);
+    }
+
+    private void OnPlayerDetached(EntityUid uid, VisionLimitComponent component, LocalPlayerDetachedEvent args)
+    {
+        _overlayMan.RemoveOverlay(_overlay);
+    }
+
+    private void OnCompInit(EntityUid uid, VisionLimitComponent component, ComponentInit args)
+    {
+        if (_player.LocalEntity == uid)
+            UpdateOverlay(component);
+    }
+
+    private void OnCompShutdown(EntityUid uid, VisionLimitComponent component, ComponentShutdown args)
+    {
+        if (_player.LocalEntity == uid)
+        {
+            _overlayMan.RemoveOverlay(_overlay);
+        }
```

Comment:
```suggestion
        if (_player.LocalEntity == uid)
            _overlayMan.RemoveOverlay(_overlay);
```
Скобочки можно убрать